### PR TITLE
GH-140: Support for Composite Discovery client

### PR DESCRIPTION
**Pull request by bijukunjummen**
#140 - Added support for a composite Discovery Client

### CODE REVIEW

File: spring-cloud-commons/src/test/java/org/springframework/cloud/client/discovery/simple/DiscoveryClientAutoConfigurationDefaultTests.java [hunk 1]
```diff
@@ -3,36 +3,40 @@
...
  * @author Biju Kunjummen
  */
 @RunWith(SpringRunner.class)
-@SpringBootTest(classes = DiscoveryClientAutoConfigurationDefaultTests.App.class)
+@SpringBootTest(classes = DiscoveryClientAutoConfigurationDefaultTests.Config.class)
 public class DiscoveryClientAutoConfigurationDefaultTests {
 
 	@Autowired
-	DiscoveryClient discoveryClient;
+	private List<DiscoveryClient> discoveryClients;
```
Comment (spencergibb) [inline spring-cloud-commons/src/test/java/org/springframework/cloud/client/discovery/simple/DiscoveryClientAutoConfigurationDefaultTests.java]: I think this test should still work as is.

**Comment by spencergibb** [comment]
Apps need to just be able to inject `DiscoveryClient` without any changes.

File: spring-cloud-commons/src/test/java/org/springframework/cloud/client/discovery/simple/DiscoveryClientAutoConfigurationDefaultTests.java (hunk 1 above)
Comment (bijukunjummen) [inline spring-cloud-commons/src/test/java/org/springframework/cloud/client/discovery/simple/DiscoveryClientAutoConfigurationDefaultTests.java]: Yes, it does @spencergibb, I have marked the `CompositeDiscoveryClient` as primary and auto-configuring the `SimpleDiscoveryClient`, a different test captures that behavior. I will restore this test to show that the defaults are maintained

**Comment by bijukunjummen** [comment]
Anything else here @spencergibb, not sure if the assumptions are right - now there are two DiscoveryClients created by default, the `CompositeDiscoveryClient` and the `SimpleDiscoveryClient`(if client defined bean does not exist). CompositeDiscoveryClient is marked `@Primary` so that it is the one injected when a client requests it.

**Comment by spencergibb** [comment]
how about a rebase @bijukunjummen?

**Comment by bijukunjummen** [comment]
Will do @spencergibb - will revisit the implementation once more, rebase and update the PR.

**Comment by bijukunjummen** [comment]
Done @spencergibb, rebased now.

**Comment by HelloDhero** [comment]
Could you help me modify this code to make the getInstances method collect all ServiceInstance objects from every DiscoveryClient?"

**Pull request merged**